### PR TITLE
fix(components/forms): implement `Validator` in the checkbox component to check required state

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
Instead of adding the `Validator.required` validator to the consumer's form control directly, implement the `Validator` interface to check if the form control satisfies the "required" and "requiredTrue" validators. We don't want to add/remove validators to/from the consumer's control since it's an unexpected side effect and could cause confusion.

NOTE: all existing tests pass.

Related: https://github.com/blackbaud/skyux/pull/1944

### CODE REVIEW

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.ts [hunk 1]
```diff
@@ -366,23 +377,31 @@ export class SkyCheckboxComponent implements ControlValueAccessor, OnInit {
...
+    this.control ||= control;
+
+    // In template-driven forms, Angular's native 'required' attribute directive only works
+    // on `input[type="checkbox"]` selectors, so we need to write the validation logic ourselves.
+    // Also, we're treating Validators.required the same as Validators.requiredTrue internally, so
+    // we need to account for that decision in our custom validator.
+    return this.isCheckboxRequired && control.value !== true
+      ? { required: true }
+      : null;
+  }
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/checkbox/checkbox.component.ts:400]: ☝️ Primary change, coupled with the removal of the `#setValidators` method, below.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2297/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/forms-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2297/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2297/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2297/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2297/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2297/playground/)

File: apps/playground/src/app/components/forms/checkbox/checkbox.component.ts [hunk 2]
```diff
@@ -79,4 +80,9 @@ export class CheckboxComponent implements OnInit {
   public toggleInlineHelp(): void {
     this.showInlineHelp = !this.showInlineHelp;
   }
+
+  protected onSubmitContactMethod(): void {
```
Comment (Blackbaud-ErikaMcVey) [inline apps/playground/src/app/components/forms/checkbox/checkbox.component.ts:84]: I realize this is just the playground but I did expect the "agree to terms and conditions" checkbox to be marked as touched as well

File: apps/playground/src/app/components/forms/checkbox/checkbox.component.ts [hunk 3]
```diff
@@ -64,7 +65,7 @@ export class CheckboxComponent implements OnInit {
 
   public ngOnInit(): void {
     this.reactiveFormGroup = this.#formBuilder.group({
-      reactiveCheckbox: new FormControl(undefined),
+      reactiveCheckbox: new FormControl(undefined, [Validators.requiredTrue]),
```
Comment (Blackbaud-ErikaMcVey) [inline apps/playground/src/app/components/forms/checkbox/checkbox.component.ts:68]: Is this an intentional change? the `required` input should be enough to make this required; the consumer shouldn't have to pass the validator. Or is this just ensuring it still works if they try to pass their own validators?

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.ts [hunk 4]
```diff
@@ -30,9 +38,19 @@ import { SkyCheckboxChange } from './checkbox-change';
...
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: SkyCheckboxComponent,
+      multi: true,
+    },
+    { provide: SKY_FORM_ERRORS_ENABLED, useValue: true },
+  ],
 })
-export class SkyCheckboxComponent implements ControlValueAccessor, OnInit {
+export class SkyCheckboxComponent
+  implements ControlValueAccessor, OnInit, Validator
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/forms/src/lib/modules/checkbox/checkbox.component.ts:52]: TIL about the `Validator` interface

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

File: apps/playground/src/app/components/forms/checkbox/checkbox.component.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply apps/playground/src/app/components/forms/checkbox/checkbox.component.ts:68]: Yes, the latter.

**Pull request merged**